src/cmap/mod.rs

```diff
@@ -216,8 +212,9 @@ impl ConnectionPoolInner {
 
         let mut available_connections = self.available_connections.lock().await;
 
-        // Close the connection if it's stale.
-        if conn.is_stale(self.generation.load(Ordering::SeqCst)) {
+        if conn.is_errored() {
```

Comment:
I assume this check occurs because connections which error when handshaking in the background thread aren't eagerly closed?

Reply:
This path is taken when a connection gets a network error after the handshake succeeds actually (e.g. during operation execution).

Reply:
Ah, right, I see now